cloud-watch-to-slack-testing/deployment/index.js

```diff
@@ -246,6 +246,29 @@ function s3ActivityMessageText(message) {
   return `${userName} generated S3 event ${eventName} from region ${awsRegion} for bucket ${bucketName} in Dockstore ${dockstoreEnvironment}`;
 }
 
+function ecsActivityMessageText(message) {
+  if (message["detail-type"] === "ECS Task State Change") {
+    return ecsTaskStateChangeMessageText(message);
+  } else {
+    return ecsAutoScalingMessageText(message);
+  }
+}
+
+function ecsTaskStateChangeMessageText(message) {
+  const taskArn = message?.detail?.taskArn;
```

Comment:
Huh, for some reason I thought optional chaining was a TypeScript only feature. Good to know it's not. Looks like the feature was added to most browsers in 2020, so when I first started on Dockstore, it actually was TS-only, and I never noticed it was added to JS.

Reply:
> Huh, for some reason I thought optional chaining was a TypeScript only feature. Good to know it's not. Looks like the feature was added to most browsers in 2020, so when I first started on Dockstore, it actually was TS-only, and I never noticed it was added to JS.

Looks like it got added to Node in version 14:
https://nodejs.medium.com/node-js-version-14-available-now-8170d384567e